sdk/src/main/java/com/mapbox/maps/Style.kt

```diff
@@ -91,6 +91,10 @@ class Style internal constructor(
    *
    * @param transitionOptions Map style transition options.
    */
+  @Deprecated(
+    "Transition options will not be applied properly, please use another function.",
+    replaceWith = ReplaceWith("mapboxMap.setStyleTransition")
+  )
   override fun setStyleTransition(transitionOptions: TransitionOptions) =
```

Comment:
What about getStyleTransition though? Should it be moved / used via MapboxMap as well?

Reply:
it's a bit tricky to implement it...
what should be returned if user did call `MapboxMap.setStyleTransition` but did not yet load a style? on the one hand we could return cached `styleTransitionOptions` but on the other hand those options are not yet applied in core. that's why I decided to leave getter for actually loaded `Style` object.

Reply:
Imo, the get/set API should appear in pairs, we can keep the `getStyleTransition` in Style object, but maybe add a `getStyleTransition` to the `MapboxMap` object and return the cached one? (similar to what `MapboxMap.setStyleTransition` does, which is set the transition to the cache which will be applied later)